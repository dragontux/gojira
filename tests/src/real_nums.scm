; check for basic correctness
(write 0.1)
(newline)

(write .10)
(newline)

(write 1.2e-2)
(newline)

(write 4.9e4)
(newline)

; operator checks
(write (* (/ 1 3.0) 3))
(newline)

(let ((foo (/ 1 3.0)))
  (write (and (< foo 0.33334)
              (> foo 0.33333))))
(newline)

(write (- 1 .5))
(newline)

(let ((foo (+ .5 (/ 1 3.0))))
  (write (and (< foo 0.83334)
              (> foo 0.83333))))
(newline)
