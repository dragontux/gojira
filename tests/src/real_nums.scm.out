0.1
0.1
0.012
49000
1
#t
0.5
#t
